asmcomp/proc_intf.ml

```diff
@@ -0,0 +1,30 @@
+module type S = sig
+  type addressing_mode
+  type specific_operation
+
+  val word_addressed: bool
+  val num_register_classes: int
+  val register_class: Reg.t -> int
+  val num_available_registers: int array
+  val first_available_register: int array
+  val register_name: int -> string
+  val phys_reg: int -> Reg.t
+  val rotate_registers: bool
+  val loc_arguments: Reg.t array -> Reg.t array * int
+  val loc_results: Reg.t array -> Reg.t array
+  val loc_parameters: Reg.t array -> Reg.t array
+  val loc_external_arguments: Reg.t array array -> Reg.t array array * int
+  val loc_external_results: Reg.t array -> Reg.t array
+  val loc_exn_bucket: Reg.t
+  val max_arguments_for_tailcalls : int
+  val safe_register_pressure: (addressing_mode, specific_operation) Mach.operation -> int
+  val max_register_pressure: (addressing_mode, specific_operation) Mach.operation -> int array
+  val destroyed_at_oper: (addressing_mode, specific_operation) Mach.instruction_desc -> Reg.t array
+  val destroyed_at_raise: Reg.t array
+  val regs_are_volatile: Reg.t array -> bool
+  val op_is_pure: (addressing_mode, specific_operation) Mach.operation -> bool
+  val num_stack_slots: int array
+  val contains_calls: bool ref
+  val assemble_file: string -> string -> int
+  val init : unit -> unit
+end
```

Comment:
This file should contain the documentation that was previously in Proc.
Note that it is often a good idea to have both an .ml and .mli file in the compiler sources to avoid parallel build problems.